.github/actions_scripts/getChangelog.pl

```diff
@@ -2,18 +2,34 @@
 
 use strict;
 use warnings;
-# get the file path
-my $ChangeLogFile = $ARGV[0];
-# get the version
-my $Version = quotemeta $ARGV[1];
-# open the file (slurp mode): https://perlmaven.com/slurp
+# get arguments
+my ($ChangeLogFile, $Version, $FormatOption) = @ARGV;
+# open the file
 open my $fh, '<', $ChangeLogFile or die;
-$/ = undef;
-my $ChangeLog = <$fh>;
-close $fh;
 
-if ( $ChangeLog =~ m/^### $Version (.*?)$(.*?)^###/sm ) {
-    print $2;
-} else {
-    print "No changelog found for this version.";
+my $SingleLineEntries = ($FormatOption or '') ne '--line-per-entry';
```

Comment:
I've just tested this and it appears to be the wrong way up.
```suggestion
my $SingleLineEntries = ($FormatOption or '') eq '--line-per-entry';
```

Reply:
Oh, I had fixed that in my working copy where I was testing, but forgot to commit it. Done now.